src/common/shibuyaHelper.js

```diff
@@ -20,12 +24,18 @@ class ShibuyaHelper {
     getBalance = async (address) => {
         const balance = (await this.api.query.system.account(address))?.data;
         return balance;
-    }
+    };
 
     createTransactExtrinsic = ({
-        targetParaId, encodedCall, fungible, requireWeightAtMost, proxyAccount, instructionWeight
+        targetParaId, encodedCall, feePerSecond, requireWeightAtMost, proxyAccount, instructionWeight,
     }) => {
-        const totalInstructionWeight = 6 * instructionWeight;
+        // The instruction count of XCM message.
+        // Because polkadotXcm.send will insert the DescendOrigin instruction at the head of the instructions list.
+        // So instructionCount should be V2.length + 1
+        const instructionCount = 6;
+        const totalInstructionWeight = instructionCount * instructionWeight;
+        const weightLimit = requireWeightAtMost + totalInstructionWeight;
+        const fungible = new BN(weightLimit).mul(new BN(feePerSecond)).div(new BN(WEIGHT_PER_SECOND));
```

Comment:
Is this fungile the same value as the hard-coded previously?

Reply:
new: 2918188416
old: 6255948005536808